agent/statemanager/state_manager_windows.go

```diff
@@ -146,11 +146,15 @@ func (manager *basicStateManager) writeFile(data []byte) error {
 		seelog.Errorf("Failed to save the data file path: %v", err)
 		return err
 	}
+	if oldFile == "" {
+		return nil
+	}
+
 	err = deps.fs.Remove(oldFile)
 	if err != nil {
 		seelog.Errorf("Error removing old file %s; err %v", oldFile, err)
+		return err
 	}
-	return nil
 }
```

Comment:
i think you still need `return nil` here.

Reply:
oops. appveyor was telling me the same thing -> `agent\statemanager\state_manager_windows.go:158: missing return at end of function
`